demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ClericalToolController.scala

```diff
@@ -0,0 +1,174 @@
+                                       conf : DemouiConfigModule,
+                                       val messagesApi: MessagesApi,
+                                       apiClient: AddressIndexClientInstance,
+                                       classHierarchy: ClassHierarchy
+                                     )(implicit ec: ExecutionContext) extends Controller with I18nSupport {
+
+  val logger = Logger("ClericalToolController")
+  val pageSize = conf.config.limit
+  val maxOff = conf.config.maxOffset
+  val maxPages = (maxOff + pageSize - 1) / pageSize
+
+  /**
+    * Present empty form for user to input address
+    *
+    * @return result to view
+    */
+  def showSingleMatchPage(): Action[AnyContent] = Action.async { implicit request =>
+    logger info ("Clerial Tool: Rendering Single Match Page")
+    val viewToRender = uk.gov.ons.addressIndex.demoui.views.html.clericalTool(
+      singleSearchForm = SingleMatchController.form,
+      warningMessage = None,
+      addressFormat = "paf",
+      pageNum = 1,
+      pageSize = pageSize,
+      pageMax = maxPages,
+      expandRow = -1,
+      addressBySearchResponse = None,
+      classification = None)
+    Future.successful(
+      Ok(viewToRender)
+    )
+  }
+
+  /**
+    * Accept posted form, deal with empty address or pass on to MatchWithInput
+    *
+    * @return result to view or redirect
+    */
+  def doMatch(): Action[AnyContent] = Action.async { implicit request =>
+    val optAddress: Option[String] = Try(request.body.asFormUrlEncoded.get("address").mkString).toOption
+    val addressText = optAddress.getOrElse("")
+    val optFormat: Option[String] = Try(request.body.asFormUrlEncoded.get("format").mkString).toOption
+    val addressFormat = optFormat.getOrElse("paf")
+    logger info ("Clerical Tool with address format = " + addressFormat)
+    if (addressText.trim.isEmpty) {
+      logger info ("Clerical Tool with Empty input address")
+      val viewToRender = uk.gov.ons.addressIndex.demoui.views.html.clericalTool(
+        singleSearchForm = SingleMatchController.form,
+        warningMessage = Some(messagesApi("single.pleasesupply")),
+        addressFormat = addressFormat,
+        pageNum = 1,
+        pageSize = pageSize,
+        pageMax = maxPages,
+        expandRow = -1,
+        addressBySearchResponse = None,
+        classification = None)
+      Future.successful(
+        Ok(viewToRender)
+      )
+    } else {
+      Future.successful(
+        Redirect(uk.gov.ons.addressIndex.demoui.controllers.routes.ClericalToolController.doMatchWithInput(addressText, addressFormat, Some(1), Some(-1)))
+      )
+    }
+  }
+
+  /**
+    * Perform match by calling API with address string. Can be called directly via get or redirect from form
+    *
+    * @param input
+    * @return result to view
+    */
+  def doMatchWithInput(input: String, formatText: String, page: Option[Int], expand: Option[Int]): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
The `formatText` may be removed in my opinion as we fetch for addresses that already contain both of the formats.